Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -367,6 +373,18 @@ export class LGServer {
     return Promise.resolve(null);
   }
 
+  private buildHoverTemplateInfo(template: LgTemplate) {
+    let templateName = '';
+    if (template.parameters.length > 0) {
+      templateName = `\\# ${template.name}(${template.parameters.join(', ')})`;
+    } else {
+      templateName = `\\# ${template.name}`;
+    }
+
+    const templateBody = template.body.replace(/-/g, '\\-').replace(/\r?\n/g, '\n\n');
```

Comment:
this should only replace all the non-escaped ```-```

Reply:
also only replace new lines at the end of each line

Reply:
I found I only need to keep the original content. So I use a code block to warp the template info here.